src/wwwroot/app/services/CandidateService.js

```diff
@@ -1,32 +1,321 @@
+   _convertRelocationPlacesToBackend(candidate);
+   _convertLanguageSkillsToBackend(candidate);
+   _convertSocialToBackend(candidate);
+   delete candidate.experienceYears;
+   delete candidate.experienceMonthes;
+   return candidate;
+}
+
+function _convertSocialToClient(candidate) {
+   let convertedSocials = [];
+   forEach(candidate.socialNetworks, social => convertedSocials.push(_getCandidateSocialObject(social)));
+   candidate.convertedSocials = convertedSocials;
+   return candidate;
+}
+
+function _convertSocialToBackend(candidate) {
+   let changedSocials = unionWith(
+      candidate.socialNetworks || [],
+      map(candidate.convertedSocials, social => {
+         return {
+            socialNetworkId  : social.socialNetwork.id,
+            path             : social.path
+         };
+      }),
+      isEquelSocials);
+
+   let deletedSocials = differenceWith(
+      changedSocials,
+      candidate.convertedSocials,
+      isEquelSocials);
+
+   forEach(deletedSocials, social => social.state = deletedState);
+   candidate.socialNetworks = changedSocials;
+   delete candidate.convertedSocials;
+   return candidate;
+}
+
+function _convertLanguageSkillsToBackend(candidate) {
+   let changedLanguageSkills = unionWith(
+      candidate.languageSkills,
+      map(candidate.convertedLanguageSkills, skill => {
+         return {
+            languageId     : skill.language.id,
+            languageLevel  : skill.languageLevel ? skill.languageLevel.id : undefined
+         };
+      }),
+      isEquelLanguageSkills);
+
+   let deletedLanguageSkills = differenceWith(
+      changedLanguageSkills,
+      candidate.convertedLanguageSkills,
+      isEquelLanguageSkills);
+
+   forEach(deletedLanguageSkills, skill => skill.state = deletedState);
+   candidate.languageSkills = changedLanguageSkills;
+   delete candidate.convertedLanguageSkills;
+   return candidate;
+}
+
+function isEquelLanguageSkills(languageSkill1, languageSkill2) {
+   return languageSkill1.languageId === languageSkill2.languageId &&
+      languageSkill1.languageLevel === languageSkill2.languageLevel;
+}
+
+function isEquelRelocationPlaces(relocationPlace1, relocationPlace2) {
+   return relocationPlace1.countryId === relocationPlace2.countryId &&
+      relocationPlace1.cityId === relocationPlace2.cityId;
+}
+
+function isEquelSocials(social1, social2) {
+   return social1.socialNetworkId === social2.socialNetworkId;
+}
+
+function _getLanguageSkillObgect(sourceLanguageSkill) {
+   let convertedLanguageSkill = {};
+   _ThesaurusService.getThesaurusTopic('language', sourceLanguageSkill.languageId)
+      .then(curriedSet(convertedLanguageSkill, 'language'));
+   _ThesaurusService.getThesaurusTopic('languageLevel', sourceLanguageSkill.languageLevel)
+      .then(level => {
+         convertedLanguageSkill.languageLevel = isNull(level) ? undefined : level;
+      });
+   return convertedLanguageSkill;
+}
+
+function _getRelocationPlaceObgect(sourceRelocationPlace) {
```

Comment:
`_getLanguageSkillObgect` and `_getLanguageSkillObgect` look like the same. maybe you can do better with refactoring